src/WebJobs.Extensions.DurableTask/Listener/TaskActorShim.cs

```diff
@@ -288,31 +364,63 @@ private async Task ProcessRequestAsync(OperationMessage request)
-            if (this.numberOperationsStarted >= OperationLimit)
+            // if there are requests waiting in the queue that we can process now, and
+            // we have not yet reached the limit, do that now; else continue as new
+
+            if (this.numberOperationsStarted < OperationLimit
+                && this.context.State.LockedBy == null
+                && this.context.State.TryDequeue(out var operationMessage))
             {
-                this.SignalContinueAsNew();
+                this.Config.TraceHelper.FunctionCompleted(
+                    this.context.HubName,
+                    this.context.Name,
+                    this.context.InstanceId,
+                    request.Id.ToString(),
+                    request.Operation,
+                    this.Config.GetIntputOutputTrace(response.Result),
+                    false,
+                    FunctionType.Actor,
+                    this.context.IsReplaying);
+
+                this.numberOperationsStarted++;
+                this.lastStartedOperation = this.ProcessRequestAsync(operationMessage);
             }
             else
             {
-                // start next request waiting in the queue if there is one
-                if (this.context.State.TryDequeue(out var operationMessage))
-                {
-                    this.numberOperationsStarted++;
-                    this.lastStartedOperation = this.ProcessRequestAsync(operationMessage);
-                }
+                this.SignalContinueAsNew();
 
-                // if there are no requests waiting, and the actor no longer exists,
-                // call ContinueAsNew. This removes the history.
-                else if (!this.context.State.ActorExists)
-                {
-                    this.SignalContinueAsNew();
-                }
+                this.Config.TraceHelper.FunctionCompleted(
+                    this.context.HubName,
+                    this.context.Name,
+                    this.context.InstanceId,
+                    request.Id.ToString(),
+                    request.Operation,
+                    this.Config.GetIntputOutputTrace(response.Result),
+                    true,
+                    FunctionType.Actor,
+                    this.context.IsReplaying);
+            }
+        }
+
+        public override void TraceAwait()
+        {
+            // we only trace the awaits that are happening inside an operation,
+            // not the awaits that come from the scheduler loop
+
+            if (this.context.CurrentOperation != null)
+            {
+                this.Config.TraceHelper.FunctionAwaited(
+                this.context.HubName,
+                this.context.Name,
+                this.context.FunctionType,
+                this.context.InstanceId,
+                this.context.CurrentOperation?.Id.ToString(),
+                this.context.CurrentOperation?.Operation,
+                this.context.IsReplaying);
```

Comment:
nit: these arguments need to be indented.

Reply:
ok.